rest/oidc_test_provider_test.go

```diff
@@ -39,3 +48,84 @@ func TestExtractSubjectFromRefreshToken(t *testing.T) {
 	require.NoError(t, err, "invalid refresh token error")
 	assert.Equal(t, subject, sub)
 }
+
+func TestInsecureOpenIDConnectAuth(t *testing.T) {
```

Comment:
Can we add a version of this test that runs with OIDCTlsSkipVerify=false (and expects error) to validate that the test is covering the scenario, and to make sure we don't introduce a regression that enables InsecureSkipVerify everywhere.

Reply:
Sure Adam, included an additional unit test that runs with OIDCTlsSkipVerify=false.